### ISSUE TITLE
Unhandled crash causing sab to get stuck on a restart loop

### ISSUE THREAD

**Issue by Drapersniper** [sabnzbd/sabnzbd#2059]
```
sabnzbd  | 2022-02-06 21:04:14,072::WARNING::[__init__:498] Restarting because of crashed assembler
sabnzbd  | Exception in thread Thread-18:
sabnzbd  | Traceback (most recent call last):
sabnzbd  |   File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
sabnzbd  |     self.run()
sabnzbd  |   File "/app/sabnzbd/assembler.py", line 184, in run
sabnzbd  |     nzo.handle_par2(nzf, filepath)
sabnzbd  |   File "/app/sabnzbd/decorators.py", line 36, in call_func
sabnzbd  |     return f(*args, **kw)
sabnzbd  |   File "/app/sabnzbd/nzbstuff.py", line 1055, in handle_par2
sabnzbd  |     pack = sabnzbd.par2file.parse_par2_file(filepath, nzf.nzo.md5of16k)
sabnzbd  |   File "/app/sabnzbd/par2file.py", line 97, in parse_par2_file
sabnzbd  |     total_size = os.path.getsize(fname)
sabnzbd  |   File "/usr/lib/python3.9/genericpath.py", line 50, in getsize
sabnzbd  |     return os.stat(filename).st_size
sabnzbd  | FileNotFoundError: [Errno 2] No such file or directory: '
```

I assume to reproduce this you should delete the par file before the repairing starts but after the download completes - not sure how it happened on my end but it creates a restart loop for sabnzbd on my container

**Comment by Drapersniper** [comment]
hmm as I pressed create the issue, it came back to life after restarting itself 8 times.

**Comment by puzzledsab** [comment]
Did you delete the file manually or use the trash can in SABnzbd?

**Comment by Drapersniper** [comment]
That's a no for both of those things.

maybe sonarr deleted it ? But I don't see it would have if the download hasn't finished yet and wouldn't have been marked as failed

**Comment by puzzledsab** [comment]
Curious. Did you exclude the file name or is that the full error message?

**Comment by Drapersniper** [comment]
Excluded, I can post it if it will help - the FileNotFound exception shouldn't really be triggered by any formatting issues.

That being said as it was printed on console it didn't look like it was invalid in Ubuntu ZFS.

**Comment by thezoggy** [comment]
Can you share the actual log output? Also what version of sab and details of how you are running it

**Comment by Safihre** [comment]
Sounds the same as #2054 and #1509.
Working on a solution.

**Comment by Drapersniper** [comment]
Sabnzbd: 3.5.1RC1 [76c5895]
Docker image: image: cr.hotio.dev/hotio/sabnzbd:testing

> Can you share the actual log output? 
I did which is the crash.

I run Sab with Error Logs only - and if an issue is not reproducible can't get debug logs.

@Safihre yeah sounds exactly like https://github.com/sabnzbd/sabnzbd/issues/2054

**Comment by Safihre** [comment]
Closed by 9ad5bd477b1a10c326f340583da2de3bfb15d132 in 3.5.1RC2.
